wgpu-core/src/hub.rs

```diff
@@ -264,6 +266,12 @@ impl<B: hal::Backend> Access<ComputePipeline<B>> for BindGroup<B> {}
 impl<B: hal::Backend> Access<RenderPipeline<B>> for Device<B> {}
 impl<B: hal::Backend> Access<RenderPipeline<B>> for BindGroup<B> {}
 impl<B: hal::Backend> Access<RenderPipeline<B>> for ComputePipeline<B> {}
+impl<B: hal::Backend> Access<QuerySet<B>> for Root {}
+impl<B: hal::Backend> Access<QuerySet<B>> for Device<B> {}
+impl<B: hal::Backend> Access<QuerySet<B>> for Sampler<B> {}
+impl<B: hal::Backend> Access<QuerySet<B>> for CommandBuffer<B> {}
+impl<B: hal::Backend> Access<QuerySet<B>> for RenderPipeline<B> {}
+impl<B: hal::Backend> Access<QuerySet<B>> for ComputePipeline<B> {}
```

Comment:
uh, that's quite a bit! can we reduce this set? I.e. can we make sure the query set is always locked after command buffers only?

Reply:
I tried and managed to get rid of one of them, but doing that would just trade edges before the query set for edges after.

Reply:
that's ok, I can try to follow-up and try to reduce this